compiler/src/dmd/inline.d

```diff
@@ -1661,6 +1661,9 @@ private bool canInline(FuncDeclaration fd, bool hasthis, bool hdrscan, bool stat
         assert(fd.semanticRun >= PASS.semantic3done);
     }
 
+    if (fd.skipCodegen)
+        return false;
```

Comment:
Is this necessary? From past experience, forcing a function to be always inlined helps with making libraries "header only" (pay as you go), similar to function templates are emitter in the compilation unit of their caller.
So one may assume that if codegen is skipped for a function than its effectively always inlined. (I understand that this is a separate matter from allowing functions that use the runtime to be called in CTFE or otherwise speculative contexts.)

Reply:
Yes. If it isn't, then GC calling code can be emitted into the object file.

The inliner happens after CTFE, so it does not affect CTFE.